modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1373 @@
+        when(AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                .thenThrow(InternalErrorException.class);
+
+        when(userManager.getMe(userName,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(userOld);
+
+        User validatedUser = (User) ServerSideValidator.validateUpdatedSCIMObject(userOld, userNew, schema);
+        when(userManager.updateMe(anyObject(), anyObject())).thenReturn(validatedUser);
+
+        //Assertions
+        Assert.assertThrows(InternalErrorException.class, () -> {
+            meResourceManager.updateWithPUT(userName, scimObjectString, null,
+                    attributes, excludeAttributes);
+        });
+
+    }
+
+    //NotFoundException
+    @Test(dataProvider = "dataForTestUpdateWithPUTExceptions")
+    public void testUpdateWithPUTNoUserExistsWithTheGivenUserName(String userName, String scimObjectString, String
+            attributes, String excludeAttributes, Object objectNEWUser, Object objectOLDUser)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        User userNew = (User) objectNEWUser;
+        User userOld = (User) objectOLDUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Me");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                .thenThrow(NotFoundException.class);
+
+        when(userManager.getMe(userName,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(null);
+
+        User validatedUser = (User) ServerSideValidator.validateUpdatedSCIMObject(userOld, userNew, schema);
+        when(userManager.updateMe(anyObject(), anyObject())).thenReturn(validatedUser);
+
+        //Assertions
+        Assert.assertThrows(NotFoundException.class, () -> {
+            meResourceManager.updateWithPUT(userName, scimObjectString, userManager,
+                    attributes, excludeAttributes);
+        });
+
+    }
+
+    // CharonException
+    @Test(dataProvider = "dataForTestUpdateWithPUTExceptions")
+    public void testUpdateWithPUTUpdatedUserResourceIsNull(String userName, String scimObjectString, String
+            attributes, String excludeAttributes, Object objectNEWUser, Object objectOLDUser)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        // userNew variable is not used here, since new user is null for this testcase
+        User userNew = (User) objectNEWUser;
```

Comment:
why is this needed? seems like `userNew` object is not used anywhere in the method

Reply:
Fixed in commit b3385c2ce740925ccf69bcb3792cd019b0c49b30